docs/source/development/technology-stack.rst

```diff
@@ -0,0 +1,150 @@
+
+The following sections describe the particular cloud services used in the SDC:
+
+CDK
+"""
+
+
+API Gateway
+"""""""""""
+
+Lambda
+""""""
+
+Batch
+"""""
+
+EventBridge
+"""""""""""
+
+RDS
+"""
+
+`RDS <https://lasp.colorado.edu/galaxy/display/IMAP/SDC+Database+Tables>`_ was
+famously chosen after an extremely lengthy discussion going back-and-forth on
+pros and cons of various database technologies (OpenSearch and DynamoDB were the
+other major considerations). RDS was ultimately chosen for being the "old
+reliable" option, as it has been extensively used and is extremely well
+documented.
+
+S3
+""
+
+
+Common Data Format
+~~~~~~~~~~~~~~~~~~
+
+The Common Data Format (CDF) was selected as the file format for IMAP data from
+requirements. CDF is a widely used data format within the Heliophsyics
+community. This decision was based purely on `requirements
+<https://lasp.colorado.edu/galaxy/display/IMAP/IMAP+SDC+to+Instrument+Team+ICD#IMAPSDCtoInstrumentTeamICD-1.3FormatStandards>`_
+and arguably would not be the top choice amongst IMAP SDC developers. As such,
+no other data formats were considered.
+
+Docker
+~~~~~~
+
+GitHub
+~~~~~~
+
+The SDC uses `GitHub <https://github.com/IMAP-Science-Operations-Center>`_ for
+version controlling its software, as well as keeping track of development tasks
+(i.e. GitHub Issues) and progress (i.e. GitHub Projects), and performing code
+reviews. GitHub was chosen over other solutions like GitLab and Bitbucket mainly
+for its collaborative features and unlimited free public repositories. As the
+IMAP SDC strives to comply with the (relatively new) `NASA SMD SPD-41a policies
+<https://smd-cms.nasa.gov/wp-content/uploads/2023/08/smd-information-policy-spd-41a.pdf>`_,
+this open-source collaborative GitHub solution made the most sense.
+
+Poetry
+~~~~~~
+
+
+Pytest
+~~~~~~
+
+The `pytest <https://docs.pytest.org>`_ python library was chosen for writing
+and executing unit tests. This library was chosen because it has a large and
+active community of users, is well documented, and has several features that
+allow for a robust testing framework (e.g. fixtures, parametrized testing,
+object mocking, etc.). There are other tools/libraries for unit testing in
+Python (e.g. ``unittest``, ``nose``, etc.), but those were not really
+considered.
+
+
+Python
+~~~~~~
+
+The SDC uses Python as the primary programming language for the implementation
+of the vast majority of the system and its ancillary tools. A few other
+languages are supported for Level 3 `algorithm development containers
+<https://github.com/IMAP-Science-Operations-Center/imap_matlab_processing_example>`_,
+and a lot of the SDC cloud infrastructure is built with AWS CDK, but otherwise
+everything is written in Python. Python was chosen mostly because at the time of
+development, it was the most widely used and supported language for scientific
+software development, and the main programming language used within the
```

Comment:
One day, *I* will be the data systems director, and then I will FORCE everyone to use Rust instead of Python! Muahahaha!! 😆